pkg/services/searchV2/extract/testdata/empty-datasource-variable-info.json

```diff
@@ -7,8 +7,8 @@
   ],
   "datasource": [
     {
-      "uid": "default.uid",
-      "type": "default.type"
+      "uid": "${sqllite}",
```

Comment:
if we keep `${sqllite}` in the output array, the dashboard preview permissions will stop working (cause no one has access to `${sqllite}`). Not sure whats the best approach here 🤔 `unresolved_datasource_variables` array? or handling this logic on a client side (permissions module)?

Reply:
🤔 -- not sure what is best here, the root question is if this is an error case or just a pattern that is hard for us to detect from the saved JSON.

In the current (not this PR) form, we will allow access to anything we do not resolve
In this PR, that would change to deny access

Maybe we should remove the UID, but keep the type?

Reply:
> 🤔 -- not sure what is best here, the root question is if this is an error case or just a pattern that is hard for us to detect from the saved JSON.

This is an "expected error case", e.g. when datasource was removed, or when the ds variable regex is overly constraining. 

I don't particularly like reusing the current `datasource` type by emptying the UID and keeping just the type, because that would require special logic in each client (is each dsRef in the array _actually_ a dsRef?). Wdyt about adding a separate `datasource_type` field?